src/zenml/utils/string_utils.py

```diff
@@ -161,16 +162,39 @@ def format_name_template(
 
     Returns:
         The formatted name template.
+
+    Raises:
+        KeyError: If a key in template is missing in the kwargs.
     """
-    kwargs["date"] = kwargs.get(
-        "date",
-        datetime.datetime.now(datetime.timezone.utc).strftime("%Y_%m_%d"),
-    )
-    kwargs["time"] = kwargs.get(
-        "time",
-        datetime.datetime.now(datetime.timezone.utc).strftime("%H_%M_%S_%f"),
-    )
-    return name_template.format(**kwargs)
+    pattern = r"\{[^}]+\}"
```

Comment:
Patter is an overkill indeed, but a safety gate to not import `get_step_context` all the time - I would keep, since it might be inefficient on frequent calls otherwise